packages/engine/src/datastore/table/state/mod.rs

```diff
@@ -42,6 +37,8 @@ pub struct State {
     removed_batches: Vec<String>,
 
     num_agents: usize,
+
+    create_parameters: StateCreateParameters,
```

Comment:
We reuse the `memory_base_id` and `message_schema` to avoid passing in the same config file again. I agree that this is weird. I changed this to store it directly instead.